In [ ]:
import pandas as pd
print("hello")

hello


## Pandas Notes
**pd.read_csv()**

used to read a csv file 


Series- 1D labeled array , can hold any data type like int, float, str or python objects. Each element in this has unique label called an index.
Dataframe- 2D labeled data structure in pandas, like a table in sql, the rows have indices and coloumns have labels.


In [ ]:
import pandas as pd
 
data ={
    'name': ['Alice', 'Bob', 'Charlie'],
    "age":[20,10,30],
    "city":["nagpur","mumbai","delhi"]
    } 
df= pd.DataFrame(data)
print(df)

df.to_csv("output.csv",index=False)


      name  age    city
0    Alice   20  nagpur
1      Bob   10  mumbai
2  Charlie   30   delhi


The dataframe is made and if we wan to save it in a file then write the last line and the type of file u want to make and if u don't want the index then write false along with the code.

Data Exploration -1.  understand the data set(rows and coloumns) 2. identify the problems 3. plan next steps .

to display rows of a dataset head(n) and tail(n) are used.If u skip the n then it prints first 5 rows by default.

In [ ]:
import pandas as pd
df= pd.read_csv(r"C:\Users\Saloni Sharma\Downloads\sales_data.csv")
print("Display first 5 rows:")
df.head(5)
print("Display last 5 rows:")
df.tail(5)

Display first 5 rows:
Display last 5 rows:


,Order_ID,Date,Region,Product,Quantity,Unit_Price_INR,Sales_INR
15,1016,2026-05-02,West,Laptop,4,55000,220000
16,1017,2026-05-09,North,Mobile Phone,8,22000,176000
17,1018,2026-05-17,South,Headphones,20,2500,50000
18,1019,2026-06-05,East,Monitor,6,15000,90000
19,1020,2026-06-18,West,Keyboard,12,1800,21600


Now to understand the data , we should now how many rows and columns are there, what type of data it is,where is missing data. 
For that we use info() method which will give a concise summary of the data like number of rows and colns, columns name , datatype,  non null counts, memory usage of the data frame.

In [ ]:
df= pd.read_csv(r"C:\Users\Saloni Sharma\Downloads\sales_data.csv")
print("Displaying the info of the data set:")
print(df.info())

Displaying the info of the data set:
<class 'pandas.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   Order_ID        20 non-null     int64
 1   Date            20 non-null     str  
 2   Region          20 non-null     str  
 3   Product         20 non-null     str  
 4   Quantity        20 non-null     int64
 5   Unit_Price_INR  20 non-null     int64
 6   Sales_INR       20 non-null     int64
dtypes: int64(4), str(3)
memory usage: 1.7 KB
None


In [ ]:
import pandas as pd
 
data ={
    'name': ['Alice', 'Bob', 'Charlie'],
    "age":[20,10,30],
    "city":["nagpur","mumbai","delhi"]
    } 
df=pd.DataFrame(data)
print("display the info of the dataset:")
print(df.info())

display the info of the dataset:
<class 'pandas.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   name    3 non-null      str  
 1   age     3 non-null      int64
 2   city    3 non-null      str  
dtypes: int64(1), str(2)
memory usage: 236.0 bytes
None


Describe Method- It provides a summary of the descriptive statistics for numerical columns in your dataframe.

In [ ]:
# step1 - sample data frame
import pandas as pd

data= { "Name":["raj","sahil","rishi","rohit", "ramesh","sachin"],
        "Age":[20,21,22,23,24,25],
        "salary":[49000,50000,56000,67000,47000,76000],
        "performance score":[90,80,45,26,37,27]
}
df=pd.DataFrame(data)
print("sample DataFrame:")
print(df)
print("descriptive statistics:")
print(df.describe())


sample DataFrame:
     Name  Age  salary  performance score
0     raj   20   49000                 90
1   sahil   21   50000                 80
2   rishi   22   56000                 45
3   rohit   23   67000                 26
4  ramesh   24   47000                 37
5  sachin   25   76000                 27
descriptive statistics:
             Age        salary  performance score
count   6.000000      6.000000           6.000000
mean   22.500000  57500.000000          50.833333
std     1.870829  11606.032914          27.549350
min    20.000000  47000.000000          26.000000
25%    21.250000  49250.000000          29.500000
50%    22.500000  53000.000000          41.000000
75%    23.750000  64250.000000          71.250000
max    25.000000  76000.000000          90.000000


The two outputs are entirely differnet. The describe method have described the data . Count-The no. of non nulls in each column.

Mean- the arithematic average of the value in each column.

Std- (standard deviation)-tells us how much of the values in a column are diff from the average(mean).

For understanding it has two types 

small std - those values which are very close to the average like [49,48,51,52] all are very close to the 50.

large std-like salaries of the employees are very differ and spread out .
Used to measure the consistency like if small std then more consistency.
Used to detect variations like if large std then means large variation in data.

Min - minimum value , smallest value in that column data.

25%- means first quarter of your data,if we sort our numbers from smallest to largest then 25% of data will be below this value . if we sort our data and divide it into 4 groups then 25% will show the min value it can be used to tell you what value separate that lowest 25% data from the rest.eg it can show the salary threshold for the lowest paid employees.eg- to calculate low paying job.

50% also called medain- the middle va;ue of your data when it was sorted. like 22+23/2=22.5 eg - to calculate medium salary paying jobs.

75% tells you about the highest 255 of your data from the rest. eg- to calculate highest paying job

Max- max value in the column.

Shape- used to know about the no. of rows and columns in your dataset. It will return a tuple with two values no. of rows and colns.
column- will give the names of the columns.

In [1]:
import pandas as pd

data= { "Name":["raj","sahil","rishi","rohit", "ramesh","sachin"],
        "Age":[20,21,22,23,24,25],
        "salary":[49000,50000,56000,67000,47000,76000],
        "performance score":[90,80,45,26,37,27]
}
df= pd.DataFrame(data)
print(df)
print('Shape:',df.shape)
print('Columns:',df.columns)

     Name  Age  salary  performance score
0     raj   20   49000                 90
1   sahil   21   50000                 80
2   rishi   22   56000                 45
3   rohit   23   67000                 26
4  ramesh   24   47000                 37
5  sachin   25   76000                 27
Shape: (6, 4)
Columns: Index(['Name', 'Age', 'salary', 'performance score'], dtype='object')


In [ ]:
print("hello")